apis/core.oam.dev/v1beta1/resourcetracker_types.go

```diff
@@ -69,9 +70,61 @@ const (
 
 // ResourceTrackerSpec define the spec of resourceTracker
 type ResourceTrackerSpec struct {
-	Type                  ResourceTrackerType `json:"type,omitempty"`
-	ApplicationGeneration int64               `json:"applicationGeneration"`
-	ManagedResources      []ManagedResource   `json:"managedResources,omitempty"`
+	Type                  ResourceTrackerType        `json:"type,omitempty"`
+	ApplicationGeneration int64                      `json:"applicationGeneration"`
+	ManagedResources      []ManagedResource          `json:"managedResources,omitempty"`
+	Compression           ResourceTrackerCompression `json:"compression,omitempty"`
+}
+
+// ResourceTrackerCompression the compression for ResourceTracker ManagedResources
+type ResourceTrackerCompression struct {
+	Type compression.Type `json:"type,omitempty"`
+	Data string           `json:"data,omitempty"`
+}
+
+// MarshalJSON .
```

Comment:
```suggestion
// MarshalJSON wil encode resource tracker spec according to the compression type, if feature enabled, it will encode data to compression data. Note: this is not actually standard json marshal process but re-use the framework function.
```

Reply:
Done